source/linux/network.c

```diff
@@ -235,6 +240,8 @@ int get_net_connections_from_proc_buf(
                     "id=%p: Could not allocate memory for connection local address",
                     (void *)ifconfig);
                 goto cleanup;
+            } else {
```

Comment:
When `if (!connection->local_interface)` is true, `connection->local_interface` is null, right? So doesn't `aws_string_destroy` handle a null pointer? It should if it doesn't and the fix should be there.

Or is it an issue of `connection` struct not initialized to zero? Together with `goto cleanup` causes deallocation of uninitialized pointer? If so, may be you want to zero the struct `connection` after line 203. And remove `has_local` and `has_remote_addr`. There is a macro `AWS_ZERO_STRUCT` that can initialize a struct to zero.

Reply:
Maybe as a debug precondition a free/delete function would check for null, but otherwise it's up to the caller to ensure it is invoked correctly.

Reply:
@fufranci I was handling the scenario where we may fail to allocate memory for the either the local interface or remote address. In those cases, the connection will not be null, and aws_string_destroy will cause a segmentation fault. I don't believe we should zero out connection, because if we fail to allocate memory for remote address for example, we still need to deallocate the memory associated with local_interface and then free memory from the connection.

Reply:
I am suggesting setting every members of `connection` to zero/null after we allocate memory for `connection`.  In such case, `connection->local_interface` is initialized to null.  And if we fail to allocation memory to `local_interface`, `local_interface` will stay as null. I think you can pass null to aws_string_destroy safely. But please double check.

So add the following after line 203 and you probably don't need the two extra bool variables.
```
AWS_ZERO_STRUCT(connection);
```

Reply:
Ah, I think you may be right. I'll make the update and test it out

Reply:
@fufranci I have updated per your recommendation! It is working, and I think it is a stylistic improvement